src/TransactionKit/TransactionKit.ts

```diff
@@ -0,0 +1,1490 @@
+      if (!provider) {
+        // Validation: if there is no provider, returns error
+        return setErrorAndReturn(
+          'Failed to get Web3 provider!',
+          'VALIDATION_ERROR',
+          {}
+        );
+      }
+
+      // Get fresh SDK instance to avoid state pollution
+      const etherspotModulaSdk = await this.etherspotProvider.getSdk(
+        this.etherspotProvider.getChainId(),
+        true
+      );
+
+      // Clear any existing operations
+      await etherspotModulaSdk.clearUserOpsFromBatch();
+
+      // Add the transaction to the userOp Batch
+      await etherspotModulaSdk.addUserOpsToBatch({
+        to: this.workingTransaction?.to || '',
+        value: this.workingTransaction?.value?.toString(),
+        data: this.workingTransaction?.data || '0x',
+      });
+
+      // Estimate the transaction
+      let estimatedUserOp;
+      try {
+        estimatedUserOp = await etherspotModulaSdk.estimate({
+          paymasterDetails,
+        });
+        this.log('Estimate userOp:', estimatedUserOp);
+      } catch (estimationError) {
+        const estimationErrorMessage = parseEtherspotErrorMessage(
+          estimationError,
+          'Failed to estimate transaction before sending.'
+        );
+        this.log('Transaction estimation before send failed', {
+          error: estimationErrorMessage,
+        });
+        return setErrorAndReturn(
+          estimationErrorMessage,
+          'ESTIMATION_ERROR',
+          {}
+        );
+      }
+
+      // Apply any user overrides to the UserOp
+      const finalUserOp = { ...estimatedUserOp, ...userOpOverrides };
+
+      // Calculate total gas cost (using the final UserOp values)
+      const totalGas = await etherspotModulaSdk.totalGasEstimated(finalUserOp);
+      const totalGasBigInt = BigInt(totalGas.toString());
+      const maxFeePerGasBigInt = BigInt(finalUserOp.maxFeePerGas.toString());
+      const cost = totalGasBigInt * maxFeePerGasBigInt;
+
+      this.log('Single transaction estimated, now sending...', {
+        to: this.workingTransaction?.to,
+        cost: cost.toString(),
+        gasUsed: totalGas.toString(),
+        userOpOverrides,
+      });
+
+      // Send the transaction
+      let userOpHash: string;
+      try {
+        userOpHash = await etherspotModulaSdk.send(finalUserOp);
+      } catch (sendError) {
+        const sendErrorMessage = parseEtherspotErrorMessage(
+          sendError,
+          'Failed to send transaction!'
+        );
+
+        this.log('Transaction send failed', { error: sendErrorMessage });
+
+        return setErrorAndReturn(sendErrorMessage, 'SEND_ERROR', {
+          to: this.workingTransaction?.to,
+          value: this.workingTransaction?.value?.toString(),
+          data: this.workingTransaction?.data,
+          chainId:
+            this.workingTransaction?.chainId ||
+            this.etherspotProvider.getChainId(),
+          cost,
+          userOp: finalUserOp,
+        });
```

Comment:
Again, great return object - gives the developer the best chance at resolving this